lib/checks/aria/allowed-role.js

```diff
@@ -0,0 +1,207 @@
+/**
```

Comment:
I think a lot of the functions and ARIA data should be made into a utility function. Probably on `axe.commons.aria`.

Reply:
Done